sdk/core/azure-core/azure/core/pipeline/transport/_base.py

```diff
@@ -220,18 +220,32 @@ class HttpRequest(object):
     :param str url: At least complete scheme/host/path
     :param dict[str,str] headers: HTTP headers
     :param files: Files list.
-    :param data: Body to be sent.
+    :param data: Body to be sent. If you want to send a json body, you should use the
+     `json` kwarg instead. We will handle json serialization for your data.
     :type data: bytes or str.
+    :keyword json: A JSON serializable object. Serializes your inputted object. Use this
+     instead of data if you wish for us to handle json serialization of your object for you.
+    :raises: TypeError
```

Comment:
Precise the circumstances in which you can get that

Reply:
I assume that `files` is yet another mutually exclusive argument, correct?

Reply:
> I assume that `files` is yet another mutually exclusive argument, correct?

This is a good question.

Now if user has 

```py
request = HttpRequest(method, url, files=xxx)
request.set_json_body(json)
```

It is valid but we will set files to None.

If we treat 
```py
request = HttpRequest(method, url, files=xxx, json=yyy)
```
as
```py
request = HttpRequest(method, url, files=xxx)
request.set_json_body(json)
```
Is it the right behavior to override `files` silently?

Reply:
@xiangyan99 following the fact that we will raise if users pass in `data` and `json`, we should raise if `files` and `json` are specified

Reply:
`files` imply multipart, correct?

Reply:
If so, yeah, while possible that one or more parts are json, I would still expect that to come in through `files`.